server/src/librelay/relay_server.go

```diff
@@ -372,7 +373,24 @@ func (relay *RelayServer) BlockCountSinceRegistration() (count uint64, err error
 		(iter.Event.Url != relay.Url) {
 		return 0, fmt.Errorf("Could not receive RelayAdded events for our relay")
 	}
-	count = lastBlockNumber - iter.Event.Raw.BlockNumber
+	blockNumber := iter.Event.Raw.BlockNumber
+
+	//Now find also last TransactionRelayed request, and use the latest of these:
+	iter1, err1 := relay.rhub.FilterTransactionRelayed(filterOpts, []common.Address{relay.Address()}, nil, nil)
+	if err1 != nil {
+		log.Println(err1)
+		return
+	}
+	// We only care about the last event
+	for iter1.Next() {
```

Comment:
Are you sure about the order when getting events?

Reply:
I don't care. I actually find the last block number of "RelayAdded", the last block of "TransactionRelayed" - and use the higher of these.
If you look at the tests, the RelayAdded is tested when creating the relay.
Then after sending a transaction, I check again, and verify that it is indeed one block into the past (the RelayAdded is 3-4 blocks down at this point)